components/org.wso2.carbon.identity.local.auth.push.authenticator/src/main/java/org/wso2/carbon/identity/local/auth/push/authenticator/PushAuthenticator.java

```diff
@@ -0,0 +1,1702 @@
+        String tenantDomain = context.getTenantDomain();
+
+        if (authenticatedUserFromContext == null) {
+
+            /*
+             * If an authenticatedUser is not found in the context, and the user is not redirected from the Identifier,
+             * redirect the user to the Identifier First page.
+             */
+            if (!isUserRedirectedFromIDF(request)) {
+                redirectUserToIDF(request, response, context);
+                context.setProperty(IS_IDF_INITIATED_FROM_AUTHENTICATOR, true);
+                return;
+            }
+
+            // If the request is returned from the Identifier First page, resolve the user and set them in context.
+            context.removeProperty(IS_IDF_INITIATED_FROM_AUTHENTICATOR);
+            AuthenticatedUser authenticatedUser = resolveUserFromRequest(request, context);
+            authenticatedUserFromContext = resolveUserFromUserStore(authenticatedUser, context);
+            setResolvedUserInContext(context, authenticatedUserFromContext);
+
+        } else if (isPreviousIdPAuthenticationFlowHandler(context)) {
+
+            /*
+             * If the previous authentication has only been done by AuthenticationFlowHandlers, need to check if the
+             * user exists in the database.
+             */
+            authenticatedUserFromContext = resolveUserFromUserStore(authenticatedUserFromContext, context);
+            setResolvedUserInContext(context, authenticatedUserFromContext);
+        }
+
+        // If the authenticated user is still null at this point, then an invalid user is trying to log in.
+        if (authenticatedUserFromContext == null) {
+            AuthenticatedUser invalidUser = new AuthenticatedUser();
+            invalidUser.setUserName(request.getParameter(USERNAME));
+            context.setProperty(IS_LOGIN_ATTEMPT_BY_INVALID_USER, true);
+            context.setProperty(INVALID_USERNAME, request.getParameter(USERNAME));
+            String randomPushAuthId = UUID.randomUUID().toString();
+            Random random = new Random();
+            redirectToPushAuthWaitPage(invalidUser, response, request, context, randomPushAuthId,
+                    Integer.toString(random.nextInt(100)));
+            return;
+        }
+
+        /*
+         * If we reach this point, a valid user is trying to log in.
+         */
+        context.removeProperty(IS_LOGIN_ATTEMPT_BY_INVALID_USER);
+        context.removeProperty(INVALID_USERNAME);
+
+        /*
+         * We need to identify the username that the server is using to identify the user. This is needed to handle
+         * federated scenarios, since for federated users, the username in the authentication context is not same as the
+         * username when the user is provisioned to the server.
+         */
+        String mappedLocalUsername = getMappedLocalUsername(authenticatedUserFromContext, context);
```

Comment:
We should only get this if the app has the config to return the associated user. Otherwise there will be a mismatch with the end identity.